client/src/components/RequestModal.js

```diff
@@ -147,42 +147,39 @@ export default class RequestModal extends Component {
         this.state = {
             alumni: null,
             studentId: '',
-            dayOptions: [],
-            day: '',
+            options: [],
+            value: '',
             topic: '',
             notes: '',
             submitting: false
         }
-        this.handleDayChange = this.handleDayChange.bind(this)
+        this.handleValueChange = this.handleValueChange.bind(this)
         this.submitRequest = this.submitRequest.bind(this)
     }
 
     async componentWillMount() {
         await this.setState({alumni: this.props.alumni})
-        this.createDayOptions(this.state.alumni.availabilities)
+        this.createOptions(this.state.alumni.availabilities)
     }
 
-    createDayOptions(availabilities) {
-        let days = [];
-        let timeOffset = new Date().getTimezoneOffset()
-        for (let day of availabilities) {
-            let adjustedTime = day.time + timeOffset
-            console.log(adjustedTime)
-            if (adjustedTime < 0) {
-                //TODO
-            } else if (adjustedTime > 2400) {
-                //TODO
-            } else {
-                if (!days.find(d => d['value'] === day.day)) {
-                    days.push({
-                        key: day.day.toLowerCase(),
-                        text: day.day,
-                        value: day.day
-                    })
-                }
-            }
-        }
-        this.setState({dayOptions: days})
+    async createOptions(availabilities) {
```

Comment:
Here is the bulk of the playing nice explained - the offset returned by JS getTimeOffset makes no sense whatsoever with our date model, and results in weird behavior - basically, I needed to make sure that 

A) Timezones were being stored relative to UTC (they weren't, but only because of offset weirdness) - FIXED
B) Because user timezones were being stored improperly, this caused a lot of otherwise functional code to break - this resulted in weirdness where after saving a preference and refreshing, the time shown would be completely different for alumni, and availabilities would be bizarre in the request modal
C) This meant weirdness when passing student timezone as a prop, which would basically double the offset in the wrong direction.

In short - it now works
I do have a question, which I will bring up during standup -
The applyTimezone in all/:timezone doesn't seem to be doing anything - this is fine, but performance wise it doesn't make sense to fix the timezone for all alumni when only a few will be requested in one session

The payload can be changed to remove the math, but to do so we will have to pass the student/alumni as a prop from App.js - that's fine, but it would also be a good first opportunity to use the store